terraform/README.md

```diff
@@ -0,0 +1,298 @@
+- **Custom Metadata Scripts** for VM initialization
+- **Ansible Playbooks** to automate post-deployment configurations
+
+This infrastructure is modular and customizable, allowing you to tailor it to specific application needs or organizational requirements.
+
+---
+
+## Pre-requisites
+
+To use this Terraform and Ansible integration, ensure you have the following tools installed:
+
+- **Google Cloud SDK** - [Installation Guide](https://cloud.google.com/sdk/docs/install)
+- **Terraform** - [Installation Guide](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+- **Ansible** - [Installation Guide](https://docs.ansible.com/ansible/latest/installation_guide/index.html)
+- **jq** - JSON processor required for handling playbook variables - [Installation Guide](https://stedolan.github.io/jq/download/)
+- **JMESPath** - [Installation Guide](https://pypi.org/project/jmespath/)
+
+Additionally, you will need:
+
+- A **Google Cloud project** with billing enabled
+- A **Service Account** with appropriate IAM roles for Compute Engine and Storage management
+
+---
+
+## Project Directory Structure
+
+The recommended project directory structure is as follows:
+
+```plaintext
+repo-root/
+├── install-oracle.sh               # Main deployment script for Ansible
+├── check-instance.yml
+├── prep-host.yml
+├── install-sw.yml
+├── config-db.yml
+├── config-rac-db.yml
+└── terraform/
+    ├── main.tf                     # Main Terraform configuration
+    ├── ansible-ssh-key             # Private SSH key file
+    ├── ansible-ssh-key.pub         # Public SSH key file
+    └── modules/
+        └── oratk-ansible/
+            ├── main.tf             # Ansible integration module
+            └── variables.tf        # Module variables
+```
+
+---
+
+## Setup and Deployment Steps
+
+1. Google Cloud Authentication
+   Authenticate using the Google Cloud SDK:
+
+```bash
+gcloud auth login
+gcloud auth application-default login
+```
+
+Set your project ID:
+
+```bash
+gcloud config set project PROJECT_ID
+```
+
+2. Generate a SSH key-pair for Ansible
+
+```bash
+ssh-keygen -t ed25519 -C "ansible" -f ansible-ssh-key -N ""
+```
+
+Ensure the correct permissions for the private key:
+
+```bash
+chmod 600 ansible-ssh-key
+```
+
+3. Review and Edit Terraform Backend Configuration
+
+   Edit `terraform/backend.tf` to define your backend settings for your state file prefix and storage bucket.
+
+   Below is an example configuration:
+
+```terraform
+terraform {
+  required_version = "1.10.5"
+
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "6.20.0"
+    }
+  }
+
+  backend "gcs" {
+    bucket = "STATE_BUCKET"
+    prefix = "STATE_PREFIX"
+  }
+}
+```
+
+4. Review and Edit Terraform Module Configuration
+
+   Edit `terraform/main.tf` to define your deployment settings. Add however many ASM disks you require in the `asm_disks` section.
+
+   Below is an example configuration:
+
+```terraform
```

Comment:
This is getting rather large to store inside documentation;  maybe store somewhere in the repo that users can just copy and use directly?